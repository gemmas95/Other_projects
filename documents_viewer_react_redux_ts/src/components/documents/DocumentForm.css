form {
  display: flex;
  flex-direction: column;
}

form button {
  align-self: center;
}

form p:first-child {
  border: 1px solid black;
  padding: 25px 10px;
  border-radius: 10px;
  text-align: center;
  background-color: #f3f3f3;
  margin: 20px;
  line-height: 1.4em;
}

form span {
  font-style: oblique;
  margin: 5px 0px;
}

input,
textarea,
select,
label {
  margin: 5px 5px 5px 5px;
  font-size: 16px;
  padding: 5px;
}

textarea {
  max-width: -webkit-fill-available;
  min-height: 100px;
  max-height: 300px;
  margin-bottom: 20px;
}
