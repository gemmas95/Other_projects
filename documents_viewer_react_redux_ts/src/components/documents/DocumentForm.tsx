import React, { ChangeEvent } from "react";
import "./DocumentForm.css";

interface Props {
  handleSubmit: any;
  handleChange: (
    event: ChangeEvent<
      HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement
    >
  ) => void;
  saving: boolean;
}

const DocumentForm = ({
  handleSubmit,
  handleChange,
  saving = false,
}: Props): JSX.Element => {
  return (
    <form autoComplete="off" onSubmit={handleSubmit}>
      <p>
        If you choose <strong>simple type</strong> of document even though you
        select an image or text, it will only be displayed the title and date.{" "}
        <br />
        For <strong>custom type</strong> will be displayed text too. <br />
        And for the <strong>advanced type</strong> all items will be displayed.
      </p>
      <label htmlFor="title">Title</label>
      <input
        id="title"
        name="title"
        type="text"
        onChange={handleChange}
        maxLength={80}
        required
      />

      <label htmlFor="type">Type</label>
      <select id="type" name="type" onChange={handleChange} required>
        <option value="">Choose a type</option>
        <option value="simple">Simple</option>
        <option value="custom">Custom</option>
        <option value="advanced">Advanced</option>
      </select>

      <label htmlFor="date">Date</label>
      <input
        id="date"
        name="date"
        type="date"
        onChange={handleChange}
        required
      />

      <label htmlFor="text">Text</label>
      <textarea id="text" name="text" onChange={handleChange} />

      <label htmlFor="image">Image</label>
      <input id="image" name="image" type="text" onChange={handleChange} />
      <p>
        The format of the image must be an url ending with a proper image
        extension(p.e: .jpeg, .jpg, gif,...) <br />
        <span>
          (example:
          https://mimusica.es/wp-content/uploads/2019/10/musica-chillout.jpg)
        </span>
      </p>

      <button className="greenButton" type="submit" disabled={saving}>
        {saving ? "Saving..." : "Create course"}
      </button>
    </form>
  );
};

export default DocumentForm;
