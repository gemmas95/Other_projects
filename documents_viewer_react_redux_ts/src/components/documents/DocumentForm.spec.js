import React from "react";
import DocumentForm from "./DocumentForm";
import { BrowserRouter } from "react-router-dom";
import { cleanup, render } from "@testing-library/react";

describe("DocumentForm", () => {
  const handleSubmit = () => {};
  const handleChange = () => {};
  let saving;

  beforeEach(() => {
    cleanup();
  });

  it("should match snapshot with saving false", () => {
    const DocumentFormTree = render(
      <BrowserRouter>
        <DocumentForm
          handleSubmit={handleSubmit}
          handleChange={handleChange}
          saving={saving}
        />
      </BrowserRouter>
    );
    expect(DocumentFormTree).toMatchSnapshot();
  });

  it("should match snapshot with saving true", () => {
    saving = true;
    const DocumentFormTree = render(
      <BrowserRouter>
        <DocumentForm
          handleSubmit={handleSubmit}
          handleChange={handleChange}
          saving={saving}
        />
      </BrowserRouter>
    );
    expect(DocumentFormTree).toMatchSnapshot();
  });
});
