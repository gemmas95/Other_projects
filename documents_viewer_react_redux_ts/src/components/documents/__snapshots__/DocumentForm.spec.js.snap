// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentForm should match snapshot with saving false 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        autocomplete="off"
      >
        <p>
          If you choose 
          <strong>
            simple type
          </strong>
           of document even though you select an image or text, it will only be displayed the title and date.
           
          <br />
          For 
          <strong>
            custom type
          </strong>
           will be displayed text too. 
          <br />
          And for the 
          <strong>
            advanced type
          </strong>
           all items will be displayed.
        </p>
        <label
          for="title"
        >
          Title
        </label>
        <input
          id="title"
          maxlength="80"
          name="title"
          required=""
          type="text"
        />
        <label
          for="type"
        >
          Type
        </label>
        <select
          id="type"
          name="type"
          required=""
        >
          <option
            value=""
          >
            Choose a type
          </option>
          <option
            value="simple"
          >
            Simple
          </option>
          <option
            value="custom"
          >
            Custom
          </option>
          <option
            value="advanced"
          >
            Advanced
          </option>
        </select>
        <label
          for="date"
        >
          Date
        </label>
        <input
          id="date"
          name="date"
          required=""
          type="date"
        />
        <label
          for="text"
        >
          Text
        </label>
        <textarea
          id="text"
          name="text"
        />
        <label
          for="image"
        >
          Image
        </label>
        <input
          id="image"
          name="image"
          type="text"
        />
        <p>
          The format of the image must be an url ending with a proper image extension(p.e: .jpeg, .jpg, gif,...) 
          <br />
          <span>
            (example: https://mimusica.es/wp-content/uploads/2019/10/musica-chillout.jpg)
          </span>
        </p>
        <button
          class="greenButton"
          type="submit"
        >
          Create course
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      autocomplete="off"
    >
      <p>
        If you choose 
        <strong>
          simple type
        </strong>
         of document even though you select an image or text, it will only be displayed the title and date.
         
        <br />
        For 
        <strong>
          custom type
        </strong>
         will be displayed text too. 
        <br />
        And for the 
        <strong>
          advanced type
        </strong>
         all items will be displayed.
      </p>
      <label
        for="title"
      >
        Title
      </label>
      <input
        id="title"
        maxlength="80"
        name="title"
        required=""
        type="text"
      />
      <label
        for="type"
      >
        Type
      </label>
      <select
        id="type"
        name="type"
        required=""
      >
        <option
          value=""
        >
          Choose a type
        </option>
        <option
          value="simple"
        >
          Simple
        </option>
        <option
          value="custom"
        >
          Custom
        </option>
        <option
          value="advanced"
        >
          Advanced
        </option>
      </select>
      <label
        for="date"
      >
        Date
      </label>
      <input
        id="date"
        name="date"
        required=""
        type="date"
      />
      <label
        for="text"
      >
        Text
      </label>
      <textarea
        id="text"
        name="text"
      />
      <label
        for="image"
      >
        Image
      </label>
      <input
        id="image"
        name="image"
        type="text"
      />
      <p>
        The format of the image must be an url ending with a proper image extension(p.e: .jpeg, .jpg, gif,...) 
        <br />
        <span>
          (example: https://mimusica.es/wp-content/uploads/2019/10/musica-chillout.jpg)
        </span>
      </p>
      <button
        class="greenButton"
        type="submit"
      >
        Create course
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DocumentForm should match snapshot with saving true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        autocomplete="off"
      >
        <p>
          If you choose 
          <strong>
            simple type
          </strong>
           of document even though you select an image or text, it will only be displayed the title and date.
           
          <br />
          For 
          <strong>
            custom type
          </strong>
           will be displayed text too. 
          <br />
          And for the 
          <strong>
            advanced type
          </strong>
           all items will be displayed.
        </p>
        <label
          for="title"
        >
          Title
        </label>
        <input
          id="title"
          maxlength="80"
          name="title"
          required=""
          type="text"
        />
        <label
          for="type"
        >
          Type
        </label>
        <select
          id="type"
          name="type"
          required=""
        >
          <option
            value=""
          >
            Choose a type
          </option>
          <option
            value="simple"
          >
            Simple
          </option>
          <option
            value="custom"
          >
            Custom
          </option>
          <option
            value="advanced"
          >
            Advanced
          </option>
        </select>
        <label
          for="date"
        >
          Date
        </label>
        <input
          id="date"
          name="date"
          required=""
          type="date"
        />
        <label
          for="text"
        >
          Text
        </label>
        <textarea
          id="text"
          name="text"
        />
        <label
          for="image"
        >
          Image
        </label>
        <input
          id="image"
          name="image"
          type="text"
        />
        <p>
          The format of the image must be an url ending with a proper image extension(p.e: .jpeg, .jpg, gif,...) 
          <br />
          <span>
            (example: https://mimusica.es/wp-content/uploads/2019/10/musica-chillout.jpg)
          </span>
        </p>
        <button
          class="greenButton"
          disabled=""
          type="submit"
        >
          Saving...
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      autocomplete="off"
    >
      <p>
        If you choose 
        <strong>
          simple type
        </strong>
         of document even though you select an image or text, it will only be displayed the title and date.
         
        <br />
        For 
        <strong>
          custom type
        </strong>
         will be displayed text too. 
        <br />
        And for the 
        <strong>
          advanced type
        </strong>
         all items will be displayed.
      </p>
      <label
        for="title"
      >
        Title
      </label>
      <input
        id="title"
        maxlength="80"
        name="title"
        required=""
        type="text"
      />
      <label
        for="type"
      >
        Type
      </label>
      <select
        id="type"
        name="type"
        required=""
      >
        <option
          value=""
        >
          Choose a type
        </option>
        <option
          value="simple"
        >
          Simple
        </option>
        <option
          value="custom"
        >
          Custom
        </option>
        <option
          value="advanced"
        >
          Advanced
        </option>
      </select>
      <label
        for="date"
      >
        Date
      </label>
      <input
        id="date"
        name="date"
        required=""
        type="date"
      />
      <label
        for="text"
      >
        Text
      </label>
      <textarea
        id="text"
        name="text"
      />
      <label
        for="image"
      >
        Image
      </label>
      <input
        id="image"
        name="image"
        type="text"
      />
      <p>
        The format of the image must be an url ending with a proper image extension(p.e: .jpeg, .jpg, gif,...) 
        <br />
        <span>
          (example: https://mimusica.es/wp-content/uploads/2019/10/musica-chillout.jpg)
        </span>
      </p>
      <button
        class="greenButton"
        disabled=""
        type="submit"
      >
        Saving...
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
